src/irmin/proof.ml

```diff
@@ -157,26 +388,54 @@ struct
         (* This phase only fills the env, it should search for anything *)
         assert false
     | Set { mode = Consume; set } ->
-        (* This is needed in order to read non-blinded nodes. *)
+        (* Use the Env to feed the values during consume *)
         Hashes.find_opt set.nodes h
+    | Stream (Produce _) ->
+        (* There is not need for sharing during in stream proofs *)
+        None
```

Comment:
Same here - calling `Env.find_node` in stream mode is probably an error? I guess using `None` also works as we don't really care...

Reply:
`find_contents` and `find_node` are called from `cached_value`, which can be used under normal operations when producing a streamed proofs.